WordPressKit/BlogServiceRemoteREST+Jetpack.swift

```diff
@@ -0,0 +1,55 @@
+public enum JetpackInstallError: String, Error {
+    case invalidCredentials = "INVALID_CREDENTIALS"
+    case forbidden = "FORBIDDEN"
+    case installFailure = "INSTALL_FAILURE"
+    case installResponseError = "INSTALL_RESPONSE_ERROR"
+    case loginFailure = "LOGIN_FAILURE"
+    case siteIsJetpack = "SITE_IS_JETPACK"
+    case activationOnInstallFailure = "ACTIVATION_ON_INSTALL_FAILURE"
+    case activationResponseError = "ACTIVATION_RESPONSE_ERROR"
+    case activationFailure = "ACTIVATION_FAILURE"
+    case unknown
+
+    init?(error code: String?) {
+        guard let code = code else {
+            self = .unknown
+            return
+        }
+
+        self.init(rawValue: code)
+    }
+}
+
+public extension BlogServiceRemoteREST {
+    public func installJetpack(url: String,
+                               username: String,
+                               password: String,
+                               completion: @escaping (Bool, JetpackInstallError?) -> Void) {
+        guard let escapedURL = url.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed) else {
+            completion(false, .unknown)
+            return
+        }
+        let path = String(format: "jetpack-install/%@/?locale=en_US", escapedURL)
```

Comment:
`?locale=en_US` I _think_ this is unnecessary (and we probably don't want it hard-coded to en_US regardless). @stevebaranski recently updated things so the locale is automatically added and set to the users preferred device language (iirc).

Reply:
Thanks for the ping @aerych!

Yes @danielebogo, if `appendsPreferredLanguageLocale` is `true` on the underlying `WordPressComRestApi` instance, this is indeed "free" behavior. Most of the changes I made to [this method](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/develop/WordPressKit/WordPressComRestApi.swift#L353) were focused on preventing duplicate parameter entries that were possible with the logic in-place.

Please let me know if you have questions or if you need additional information.

Reply:
Thanks @aerych and @stevebaranski for the hint 😊